### PR TITLE
Bug 238 d6t qty field can now be set to 0

### LINKED ISSUE #238: Quantity field in D6T table does not allow deleting the entire field
Thanks for submitting an issue! Below are a few things you can do to help us more quickly address the issue.
## Checklist
I have…
- [x] described in detail below both the behavior I expected and the behavior I observed.
- [x] thoroughly outlined below the steps to reproduce this issue, including relevant technical details (e.g. device, operating system, browser, browser version).
~~[ ] attached screenshots illustrating relevant behavior.~~

## Expected Behavior
User can enter any number in the quantity field.

## Observed Behavior
Quantity field requires a valid number at all times, and so deleting the value entirely and typing a new quantity is not possible.

Recommend switching input from a text input to a number input and adding additional unit tests.

## Steps to Reproduce
To reproduce this issue…
1. Attempt to delete quantity and type a different number in the D6T table

## Screenshots
_Attach relevant screenshots here._

### PR DESCRIPTION
**Pull request by scottpiercewDDS**
Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist
I have…
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] attached screenshots illustrating relevant behavior before and after my changes.
- ~~[ ] review and update SSAG documentation if needed.~~

## Summary of Changes
This pull request…
- Closes #238 
- Allows users to change the D6T quantity field by setting a blank input to zero

## Testing
To verify the changes proposed in this pull request…
1. Front-end unit tests
2. On the Enter Receipt (D6T), change the quantity field of a SDN

## Screenshots
image

### CODE REVIEW

File: frontend/src/pages/EnterReceiptPage/QuantityInput.tsx [hunk 1]
```diff
@@ -11,8 +11,9 @@ const QuantityInput: React.FC<QuantityInputProps> = ({
   onQuantitiyChange
 }) => {
   const onValueChange: React.ChangeEventHandler<HTMLInputElement> = e => {
-    const asNum = parseInt(e.currentTarget.value);
-    if (!isNaN(asNum)) {
+    const asNum =
+      e.currentTarget.value === "" ? 0 : parseInt(e.currentTarget.value);
```
Comment (vcastro-DDS) [inline frontend/src/pages/EnterReceiptPage/QuantityInput.tsx:15]: Can we change e to element or something similar for easier readability?

**Review by vcastro-DDS — COMMENTED** [review]
Left comment on code about variable naming.

File: frontend/src/pages/EnterReceiptPage/QuantityInput.tsx (hunk 1 above)
Comment (scottpiercewDDS) [reply frontend/src/pages/EnterReceiptPage/QuantityInput.tsx:15]: We could change it to event. @jhackshaw-dds what naming convention have you been using for input events?

Comment (jhackshaw-dds) [reply frontend/src/pages/EnterReceiptPage/QuantityInput.tsx:15]: I think `e` is fine for now as I don't think it's consistent across the rest of the code base

**Review by jhackshaw-dds — APPROVED** [review]

**Pull request merged**